crates/tasks/src/lib.rs

```diff
@@ -190,14 +191,17 @@ impl TaskManager {
     pub fn new(handle: Handle) -> Self {
         let (panicked_tasks_tx, panicked_tasks_rx) = unbounded_channel();
         let (signal, on_shutdown) = signal();
-        Self {
+        let manager = Self {
             handle,
             panicked_tasks_tx,
             panicked_tasks_rx,
             signal: Some(signal),
             on_shutdown,
             graceful_tasks: Arc::new(AtomicUsize::new(0)),
-        }
+        };
+        let _ = GLOBAL_EXECUTOR.set(manager.executor());
```

Comment:
if this errors we want to log this, for this we can use inspect_err here I believe